src/tiled/objectsdock.cpp

```diff
@@ -368,11 +368,6 @@ void ObjectsView::selectionChanged(const QItemSelection &selected,
             selectedObjects.append(o);
     }
 
-    // Switch the current object layer if only one object layer (and/or its objects)
-    // are included in the current selection.
-    if (singleObjectGroup)
-        mMapDocument->setCurrentLayer(singleObjectGroup);
```

Comment:
Since `singleObjectGroup` is no longer actually used now, you can clean up the code above.